### PR TITLE
Arizona Military Retirement Subtraction

### LINKED ISSUE #3007: Arizona Military Retirement Subtraction
https://www.azleg.gov/ars/43/01022.htm (26)

### PR DESCRIPTION
**Pull request by XZ-mabel**
Fixes #3007

### Summary
🎖️📄🌵

This change adds a new parameter file for the `military_benefits_subtraction` in Arizona income tax. The file specifies the amount of military retirement income that can be excluded from taxation for eligible individuals.

> _`military_benefits`_
> _Subtract from taxable income_
> _Autumn leaves retire_

### Walkthrough
*  Add a new parameter file for the military benefits subtraction in Arizona income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/3008/files?diff=unified&w=0#diff-8ecb8dab1759e5da3faac3dcfaa748d8e767544d3cce903833e7830d6cfc2f38R1-R14))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction:1]: not a yaml file

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  2018-01-01: 2_500
+  2019-01-01: 3_500
+  2021-01-01: .inf
+
+metadata:
+  unit: currency-USD
+  period: year
+  label: Arizona military retirement subtraction max amount
+  reference:
+    - title: 43-1022(26)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction:12]: more descriptive title + tax form

File: policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Arizona Military Retirement Subtraction"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income
+        military_retirement_pay = tax_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py:16]:
```suggestion
        head = person("is_tax_unit_spouse",
        spouse - same 
        head_or_spouse = head | spouse
        military_retirement_pay = person(
```

File: policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            "military_retirement_pay", period
+        )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py:18]:
```suggestion
        )
        eligible_military_pay = military_retirement_pay * head_or_spouse 
```

File: policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        return min_(military_retirement_pay, p.military_retirement_subtraction)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py:19]:
```suggestion
        return min_(tax_unit.sum(eligible_military_pay), p.military_retirement_subtraction)
```

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:1]: add changelog_entry

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+metadata:
+  unit: currency-USD
+  period: year
+  label: Arizona military retirement subtraction max amount
+  reference:
+    - title: Subtractions from Arizona gross income 43-1022(26) 
+      href: https://www.azleg.gov/ars/43/01022.htm
+    - title: 2021 Form 140 - Arizona Resident Personal Income Tax Booklet, page 27, 29b
+      href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:15]: add pages for all pdf files

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  2021-01-01: .inf
+
+metadata:
+  unit: currency-USD
+  period: year
+  label: Arizona military retirement subtraction max amount
+  reference:
+    - title: Subtractions from Arizona gross income 43-1022(26) 
+      href: https://www.azleg.gov/ars/43/01022.htm
+    - title: 2021 Form 140 - Arizona Resident Personal Income Tax Booklet, page 27, 29b
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:14]:
```suggestion
    - title: 2021 Form 140 - Arizona Resident Personal Income Tax Booklet, 29b
```

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+      href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=27
+    - title: 2022 Form 140 - Arizona Resident Personal Income Tax Booklet, page 25, 29b
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:16]:
```suggestion
    - title: 2022 Form 140 - Arizona Resident Personal Income Tax Booklet, 29b
```

File: policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+from policyengine_us.model_api import *
+
+
+class az_military_retirement_subtraction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona Military Retirement Subtraction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/az_military_retirement_subtraction.py:7]:
```suggestion
    label = "Arizona military retirement subtraction"
```

File: changelog_entry.yaml [hunk 9]
```diff
@@ -1,4 +1,4 @@
 - bump: minor
   changes:
     added:
-    - Kentucky family size tax credit.
+    - Arizona Military Retirement Subtraction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Arizona military retirement subtraction.
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:1]: can you a the folder structure:
subtractions - military_retirement - max_amount.yaml

File: policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  period: year
+  label: Arizona military retirement subtraction max amount
+  reference:
+    - title: Subtractions from Arizona gross income 43-1022(26) 
+      href: https://www.azleg.gov/ars/43/01022.htm
+    - title: 2021 Form 140 - Arizona Resident Personal Income Tax Booklet, 29b
+      href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2021_140BOOKLET.pdf#page=27
+    - title: 2022 Form 140 - Arizona Resident Personal Income Tax Booklet, 29b
+      href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140BOOKLET.pdf#page=25
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/military_retirement_subtraction.yaml:18]: delete extra line

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_retirement/max_amount.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+description: Arizona allows for this maximum military retirement subtraction for each eligible individual.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_retirement/max_amount.yaml:1]:
```suggestion
description: Arizona caps the military retirement subtraction at this amount for each eligible individual.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Also add a `subtractions.yaml` and add this to it

**Pull request merged**